java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -2539,6 +2539,31 @@ public final Table stringSplit(String pattern, int limit, boolean splitByRegex)
     return new Table(stringSplit(this.getNativeView(), pattern, limit, splitByRegex));
   }
 
+  /**
+   * Returns a list of columns by splitting each string using the specified regex program. The
+   * number of rows in the output columns will be the same as the input column. Null entries
+   * are added for a row where split results have been exhausted. Null input entries result in
+   * all nulls in the corresponding rows of the output columns.
+   *
+   * @param regexProg the regex program with UTF-8 encoded string identifying the split pattern
+   *                  for each input string.
+   * @param limit the maximum size of the list resulting from splitting each input string,
+   *              or -1 for all possible splits. Note that limit = 0 (all possible splits without
+   *              trailing empty strings) and limit = 1 (no split at all) are not supported.
+   * @param splitByRegex a boolean flag indicating whether the input strings will be split by a
+   *                     regular expression pattern or just by a string literal delimiter.
+   * @return list of strings columns as a table.
+   */
+  public final Table stringSplitRegexProg(RegexProgram regexProg, int limit, boolean splitByRegex) {
```

Comment:
There are some problems with this API that I wasn't happy with in the original API, but are even worse here.

Can we split this into two separate APIs instead of having a boolean flag to distinguish between them?

```
public final Table stringSplit(String delimiter, int limit) 
public final Table stringSplit(String delimiter)
```

Which will just do what they do today, splits strings on a fixed delimiter (no Regexp) using the given limit.  I am using limit because it matches the existing API, but it really would have been nice to use the same value that CUDF uses `maxSplit`, but that is separate.

The we add in the APIs

```
public final Table stringSplit(RegexProgram delimiter, int limit) 
public final Table stringSplit(RegexProgram delimiter)
```

These would split the input data on the regular expression.  Because there are new APIs it would be nice to use maxSplit, but for consistency I think we should keep the limit.

We then should deprecate all of the existing APIs that use the boolean `splitByRegexp`, and implement them in terms of the non-deprecated APIs. i.e.

```
  @Deprecated
  public final Table stringSplit(String pattern, boolean splitByRegex) {
    if (splitByRegex) {
      RegexpProgram program = new RegexpProgram(pattern, RegexpFlags.DEFAULT, CaptureGroups.NON_CAPTURE);
      return splitString(program);
    } else {
      return stringSplit(pattern);
    }
  }
```